tracer/src/Datadog.Trace/Iast/SourceType.cs

```diff
@@ -37,4 +37,21 @@ internal static string GetString(SourceTypeName value)
             SourceTypeName.CookieValue => "http.cookie.value",
             _ => string.Empty
         };
+
+    internal static SourceTypeName FromString(string? value)
+        => value switch
+        {
+            "http.request.parameter" => SourceTypeName.RequestParameterValue,
+            "http.request.parameter.name" => SourceTypeName.RequestParameterName,
+            "http.request.header" => SourceTypeName.RequestHeader,
+            "http.request.header.name" => SourceTypeName.RequestHeaderName,
+            "http.request.path" => SourceTypeName.RequestPath,
+            "http.request.body" => SourceTypeName.RequestBody,
+            "http.request.query" => SourceTypeName.RequestQuery,
+            "http.request.path.parameter" => SourceTypeName.RoutedParameterValue,
+            "http.request.matrix.parameter" => SourceTypeName.MatrixParameter,
+            "http.cookie.name" => SourceTypeName.CookieName,
+            "http.cookie.value" => SourceTypeName.CookieValue,
+            _ => default(SourceTypeName)
```

Comment:
maybe you wanna throw or have a warning of some type if value is not expected?

Reply:
Maybe, but that switch construct is so beautiful...